wpilibj/src/main/java/edu/wpi/first/wpilibj/experimental/command/CommandScheduler.java

```diff
@@ -0,0 +1,472 @@
+   *
+   * @param button the button to add
+   */
+  public void addButton(Runnable button) {
+    m_buttons.add(button);
+  }
+
+  /**
+   * Removes all button bindings from the scheduler.
+   */
+  public void clearButtons() {
+    m_buttons.clear();
+  }
+
+  /**
+   * Schedules a command for execution.  Does nothing if the command is already scheduled.
+   * If a command's requirements are not available, it will only be started if all the commands
+   * currently using those requirements have been scheduled as interruptible.  If this is
+   * the case, they will be interrupted and the command will be scheduled.
+   *
+   * @param interruptible whether this command can be interrupted
+   * @param command       the command to schedule
+   */
+  @SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.NPathComplexity"})
+  private void schedule(boolean interruptible, Command command) {
+    if (CommandGroupBase.getGroupedCommands().contains(command)) {
+      throw new IllegalUseOfCommandException(
+          "A command that is part of a command group cannot be independently scheduled");
+    }
+
+    //Do nothing if the scheduler is disabled, the robot is disabled and the command doesn't
+    //run when disabled, or the command is already scheduled.
+    if (m_disabled
+        || (RobotState.isDisabled() && !command.runsWhenDisabled())
+        || m_scheduledCommands.containsKey(command)) {
+      return;
+    }
+
+    Set<Subsystem> requirements = command.getRequirements();
+
+    //Schedule the command if the requirements are not currently in-use.
+    if (Collections.disjoint(m_requirements.keySet(), requirements)) {
+      command.initialize();
+      CommandState scheduledCommand = new CommandState(interruptible);
+      m_scheduledCommands.put(command, scheduledCommand);
+      for (Consumer<Command> action : m_initActions) {
+        action.accept(command);
+      }
+      for (Subsystem requirement : requirements) {
+        m_requirements.put(requirement, command);
+      }
+    } else {
+      //Else check if the requirements that are in use have all have interruptible commands,
+      //and if so, interrupt those commands and schedule the new command.
+      boolean allInterruptible = true;
+      for (Subsystem requirement : requirements) {
+        if (m_requirements.keySet().contains(requirement)) {
+          allInterruptible &=
+              m_scheduledCommands.get(m_requirements.get(requirement)).isInterruptible();
+        }
+      }
+      if (allInterruptible) {
+        for (Subsystem requirement : requirements) {
+          if (m_requirements.containsKey(requirement)) {
+            cancel(m_requirements.get(requirement));
+          }
+        }
+        command.initialize();
+        CommandState scheduledCommand = new CommandState(interruptible);
+        m_scheduledCommands.put(command, scheduledCommand);
+        for (Consumer<Command> action : m_initActions) {
+          action.accept(command);
+        }
+      }
```

Comment:
Should this instance be adding the command to the requirements map as well, just like the disjoint case above?

Reply:
Fixed.